include/CoreDisTools/coredistools.h

```diff
@@ -0,0 +1,52 @@
+//===--------- coredistools.h - Dissassembly tools for CoreClr
+//-------------===//
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// \brief Core Disassembly Tools API for CoreClr AOT/JIT
+///
+//===----------------------------------------------------------------------===//
+
+#if !defined(LLILC_TOOLS_COREDISTOOLS)
+#define LLILC_TOOLS_COREDISTOOLS
+
+#include <stdint.h>
+
+#if defined(__cplusplus)
+#define EXTERN_C extern "C"
+#else
+#define EXTERN_C
+#endif // defined(__cplusplus)
+
+#if defined(_MSC_VER)
+#if defined(DllInterfaceExporter)
+#define DllIface EXTERN_C __declspec(dllexport)
+#else
+#define DllIface EXTERN_C __declspec(dllimport)
+#endif // defined(DllInterfaceExporter)
+#else
+#define DllIface
+#endif // defined(_MSC_VER)
+
+enum TargetArch {
+  Target_Host, // Target is the same as host architecture
+  Target_X86,
+  Target_X64,
+  Target_Thumb,
+  Target_Arm64
+};
+
+struct CorDisasm;
+DllIface CorDisasm *InitDisasm(enum TargetArch Target);
+DllIface size_t DisasmInstruction(const CorDisasm *Disasm, size_t Address,
+                                  const uint8_t *Bytes, size_t Maxlength);
+DllIface void PrintInstruction(const CorDisasm *Disasm, size_t Address,
+                               const uint8_t *Bytes, size_t Maxlength);
+DllIface void FinishDisasm(const CorDisasm *Disasm);
+
+#endif // defined(LLILC_TOOLS_COREDISTOOLS)
```

Comment:
Nit. The comment is about the negative definition -- !defined(LLILC_TOOLS_COREDISTOOLS).